extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1174,11 +1178,27 @@ export class AmpA4A extends AMP.BaseElement {
             'src': srcPath,
             'name': nameData,
           }, SHARED_IFRAME_PROPERTIES));
+      if (method == XORIGIN_MODE.NAMEFRAME) {
+        const attributes = this.generateSentinelAndContext(iframe);
```

Comment:
I could do that, but there are a lot of other places that would have to change. could I do a cleanup refactoring PR for that? really need to finish this asap